### PR TITLE
Make sure that all the excluded collection detected even using local cache

### LINKED ISSUE #7976: Asset cache works incorrectly when change bootstrap collection via settings file
**Describe the bug (REQUIRED)**
Bug appeared when we have following setup:
* Have 2 collections. Both contains simple go with collectionproxy component. One of collection contains go with collection proxy "**Exclude**" flag (**/assets/main_lu.collection**), another - without "**Exclude**" flag (**/assets/main.collection**).
* Have **test.ini** settings file which contains reference to **/assets/main_lu.collection** as bootstrap collection.
* Use local cache (**--resource-cache-local** flag)

Initially we have clean environment.
If cache was produced from the build that didn't have any excluded assets the second build also doesn't exclude any assets because of local cache.  
⚠️  **Important**: such behavior observed in case if we change bootstrap collection via settings file. 

**To Reproduce (REQUIRED)**
1. Unzip example project
2. Build project with command 
```
java -jar bob.jar --with-symbols --bundle-output $(pwd)/output --platform arm64-ios --variant debug --archive --strip-executable --texture-compression true --liveupdate yes --use-async-build-server --resource-cache-local $(pwd)/local_cache distclean build bundle
```
3. Build project with command 
```
java -jar bob.jar --with-symbols --bundle-output $(pwd)/output --platform arm64-ios --variant debug --archive --strip-executable --texture-compression true --liveupdate yes --use-async-build-server --resource-cache-local $(pwd)/local_cache --settings test.ini distclean build bundle
```
4. Compare two zip archives with excluded assets. After the first run we get archive with liveupdate manifest only (correct behavior). After the second run we also get archive with liveupdate manifest only (incorrect behavior because we use /assets/main_lu.collection as bootstrap collection and all assets should be excluded).

**Expected behavior (REQUIRED)**
After the second build we'll get zip archive with excluded assets.

**Defold version (REQUIRED):**
  - Version: 1.5.0 (but reproduced since 1.4.4)

**Platforms (REQUIRED):**
 - Platforms: any
 - OS: any
 - Device: any

**Minimal repro case project (OPTIONAL):**
[liverunner.zip](https://github.com/defold/defold/files/12560926/liverunner.zip)

### PR DESCRIPTION
**Pull request by AGulev**
Fixed issue when using local cache may affect detection of the excluded collection proxies for LiveUpdate.

Fix https://github.com/defold/defold/issues/7976

## PR checklist

* [ ] Code
	* [ ] Add engine and/or editor unit tests.
	* [x] New and changed code follows the overall code style of existing code
	* [ ] Add comments where needed
* [ ] Documentation
	* [ ] Make sure that API documentation is updated in code comments
	* [ ] Make sure that manuals are updated (in github.com/defold/doc)
* [ ] Prepare pull request and affected issue for automatic release notes generator
	* [x] Pull request - Write a message that explains what this pull request does. What was the problem? How was it solved? What are the changes to APIs or the new APIs introduced? This message will be used in the generated release notes. Make sure it is well written and understandable for a user of Defold.
	* [x] Pull request - Write a pull request title that in a sentence summarises what the pull request does. Do not include "Issue-1234 ..." in the title. This text will be used in the generated release notes.
	* [x] Pull request - Link the pull request to the issue(s) it is closing. Use on of the [approved closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).
	* [x] Affected issue - Assign the issue to a project. Do not assign the pull request to a project if there is an issue which the pull request closes.
	* [ ] Affected issue - Assign the "breaking change" label to the issue if introducing a breaking change.
	* [ ] Affected issue - Assign the "skip release notes" is the issue should not be included in the generated release notes.

### CODE REVIEW

File: com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/Project.java [hunk 1]
```diff
@@ -2102,12 +2102,4 @@ public void setTextureProfiles(TextureProfiles textureProfiles) {
         this.textureProfiles = textureProfiles;
     }
 
-    public void excludeCollectionProxy(String path) {
-        state.addExcludedCollectionProxy(path);
-    }
-
-    public final List<String> getExcludedCollectionProxies() {
-        return state.getExcludedCollectionProxies();
-    }
```
Comment (AGulev) [inline com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/Project.java:2111]: We don't need it anymore in Project

File: com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/State.java [hunk 2]
```diff
@@ -34,9 +34,8 @@
  */
 public class State implements Serializable {
 
-    private static final long serialVersionUID = -275410118302470802L;
+    private static final long serialVersionUID = -275410118302470802L + 1;
```
Comment (AGulev) [inline com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/State.java:37]: it's easier to track version using "+N"

File: com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/GameProjectBuilder.java [hunk 3]
```diff
@@ -374,17 +376,28 @@ private static void buildResourceGraph(Project project, IResource resource, Reso
                 throw new CompileExceptionError(resource, 0, "Unable to find resource " + resource.getPath());
             }
             builder.mergeFrom(content);
-            Object message = builder.build();
-            buildResourceGraph(project, (Message) message, currentNode, visitedNodes);
+            Message message = (Message)builder.build();
+            buildResourceGraph(project, message, currentNode, visitedNodes, excludedResources);
+            if (isCollelctionProxy && excludedResources != null) {
+                CollectionProxyDesc desc = (CollectionProxyDesc)message;
+                if (desc.getExclude()) {
```
Comment (AGulev) [inline com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/GameProjectBuilder.java:383]: ... so I just check if it's excluded collection proxy or not and exclude it if needed

File: com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/GameProjectBuilder.java [hunk 4]
```diff
@@ -344,13 +345,14 @@ private static void buildResourceGraph(Project project, Message node, ResourceNo
...
-    private static void buildResourceGraph(Project project, IResource resource, ResourceNode parentNode, Collection<String> visitedNodes) throws CompileExceptionError {
+    private static void buildResourceGraph(Project project, IResource resource, ResourceNode parentNode, Collection<String> visitedNodes, List<String> excludedResources) throws CompileExceptionError {
         if (resource.getPath().equals("") || visitedNodes.contains(resource.output().getAbsPath())) {
             return;
         }
-
+        boolean isCollelctionProxy = false;
         if (resource.output().getPath().endsWith(".collectionproxyc")) {
             visitedNodes = new HashSet<String>();
+            isCollelctionProxy = true;
```
Comment (JCash) [inline com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/GameProjectBuilder.java:355]: typo

**Review by JCash — DISMISSED** [review]
typos

**Pull request merged**